app/api/db/models/base.py

```diff
@@ -60,25 +56,36 @@ def copy(self, **kwargs: dict[str, Any]) -> "Base":
         return copy
 
 
-def uuid_gen() -> uuid.UUID:
-    return uuid.uuid4()
+@declarative_mixin
+class IdMixin:
+    """Mixin to add a UUID id primary key column to a model
+    https://docs.sqlalchemy.org/en/14/orm/declarative_mixins.html
+    """
 
+    id: uuid.UUID = Column(postgresql.UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
 
-def utc_timestamp_gen() -> datetime:
-    """Generate a tz-aware timestamp pinned to UTC"""
-    return utcnow()
+
+def same_as_created_at(context: Any) -> Any:
+    return context.get_current_parameters()["created_at"]
 
 
 @declarative_mixin
 class TimestampMixin:
+    """Mixin to add created_at and updated_at columns to a model
```

Comment:
I wonder if this mixin in particular should be a class that all others get derived from using SQLAlchemy's abstract tables (or add it to Base) as I can't any scenarios where we wouldn't want to have these timestamps on a table.

Reply:
yeah i can't think of anything either but also worried about what i don't know i don't know.

what about tables that aren't used by the transactional API, like tables used in data pipelines, staging tables, and such?

Reply:
actually what about linking tables for many to many relationships? it seems like we don't always need created_at / updated_at for those

Reply:
another idea could be to create a `Resource` base class, referring to what we would think of as REST API resources. all of those should have unique ids for sure. i wonder if there are other things that need ids that aren't resources.

Reply:
A many-to-many table having the timestamps is potentially useful if the relationship is created at a different time than the related objects, but that's a very niche scenario for being necessary.

Staging tables that are temporary (eg load, do some operations, and delete) wouldn't have a ton of use as you'd never see them / the logs or derived tables associated with the processes would be good enough for the timestamps.

Definitely could see some sort of resource base class for this.